Shell/LineEditor.cpp

```diff
@@ -354,15 +388,43 @@ String LineEditor::get_line(const String& prompt)
                 }
 
                 String token = is_empty_token ? String() : String(&m_buffer[token_start], m_cursor - (size_t)token_start);
+                Vector<String> suggestions;
 
                 if (is_first_token)
-                    tab_complete_first_token(token);
+                    suggestions = tab_complete_first_token(token);
                 else
-                    tab_complete_other_token(token);
+                    suggestions = tab_complete_other_token(token);
+
+                if (m_times_tab_pressed > 1 && !suggestions.is_empty()) {
+                    size_t longest_name = 0;
+
+                    for (auto& suggestion : suggestions)
+                        longest_name = max(longest_name, suggestion.length());
+
+                    size_t num_printed = 0;
+                    putchar('\n');
+                    for (auto& suggestion : suggestions) {
+                        int new_pos = num_printed + suggestion.length() + (longest_name + 2 - suggestion.length());
+
+                        if (new_pos > m_num_columns) {
+                            putchar('\n');
+                            num_printed = 0;
+                        }
 
+                        num_printed += printf("%-*s", static_cast<int>(longest_name) + 2, suggestion.characters());
```

Comment:
Shouldn't suggestions be printed to stderr?

Reply:
Is there any particular reason this would be the case? I was the under the impression that was for errors only?

Reply:
> I was the under the impression that was for errors only?

No, that's just some misleading naming. stderr is for everything that isn't the primary output (which goes to stdout), so logging, diagnostic messages, suggestions — and in case of a shell (or any REPL), prompts, echoing stuff the user types (the commands) and suggestions.

Think if it this way: you should be able to redirect Shell's output to a file,
```sh
$ sh > output.txt
```
and still he able to use it and interact with it as normal, except programs primary output would go into the file. But all the prompts, suggestions, etc would stay.